### ISSUE TITLE
vue-docgen-api gives Cannot read property 'leadingComments' of null warning

### ISSUE THREAD

**Issue by mattgreenfield** [vue-styleguidist/vue-styleguidist#753]
**Current behavior**
Using storybook addon-docs (vue-docgen-api version 4.8.0).
When adding a the following code to a render function:
```
        if (this.$slots.default) {
            this.$slots.default.forEach(() => {
              console.log('foo');
            })
        }
```
See https://github.com/mattgreenfield/storybook-vue-sandbox/blob/leadingComments-warning/src/components/Heading/Heading.vue#L19-L23

I get this warning:
```
Module Warning (from ./node_modules/vue-docgen-loader/lib/index.js):
[vue-docgen-loader] failed to parse the component file with docgen-api: Cannot read property 'leadingComments' of null
```
See https://github.com/mattgreenfield/storybook-vue-sandbox/blob/leadingComments-warning/src/components/Button/Button.vue#L19-L23

I don't get this warning if I put the same code in a computed property for a non-render component - see 

**To reproduce**
- Clone this repo https://github.com/mattgreenfield/storybook-vue-sandbox
- Checkout to the `leadingComments-warning` branch
- `npm i && npm run storybook:serve`

**Expected behavior**
No warnings, there are no comments to extract.

**Comment by elevatebart** [comment]
Thanks @mattgreenfield 

This is a bug and I shall fix it.

Just out of curiosity, I have never seen this pattern before. 
What are you trying to loop through in this expression?

**Comment by mattgreenfield** [comment]
Thanks @elevatebart - that's great.

Here's my use case:
I have a 'Tabs' component where I loop through each element in the default slot and wrap it in a tab.
```html
<tabs>
  <div tab-title="One">This is the content of the tab 'One'</div>
  <div tab-title="Two">This is the content of the tab 'Two'</div>
</tabs>
```
Then (off the top of my head), I do something like

```js
// Tabs.vue
import TabHeader from './TabHeader.vue';
import TabItem from './TabItem.vue';

let tabHeadings = []; // as VNode[]
let tabContents = []; // as VNode[]

if (this.$slots.default) {
    this.$slots.default.forEach((element) => {
          tabHeadings.push(createElement('TabHeader', {}, [element.data.attrs.tabTitle]));
          tabContents.push(createElement('TabItem', {}, [element]));
     })
}

return createElement(
      'div',
      {},
      [tabHeadings, tabContents],
);
```

There's also some code to add the aria attributes that link up the tab title to it's content.

I hope that makes sense! I'd love to hear how you would tackle the problem or could improve it.
